datashare-app/src/main/java/org/icij/datashare/web/DocumentResource.java

```diff
@@ -174,279 +152,245 @@ public Payload searchOccurrences(
+                                            @StringToClassMapItem(key = "docIds", value = List.class),
+                                            @StringToClassMapItem(key = "tags", value = List.class)
+                                    }
+                            ),
+                            examples = {@ExampleObject(value = "{\"docIds\": [\"bd2ef02d39043cc5cd8c5050e81f6e73c608cafde339c9b7ed68b2919482e8dc7da92e33aea9cafec2419c97375f684f\", \"7473df320bee9919abe3dc179d7d2861e1ba83ee7fe42c9acee588d886fe9aef0627df6ae26b72f075120c2c9d1c9b61\"], \"tags\": [\"foo\", \"bar\"]}")}
+                    ))
+    )
+    @ApiResponse(responseCode = "200")
     @Post("/:project/documents/batchUpdate/untag")
     public Payload groupUntagDocument(final String projectId, BatchTagQuery query,  Context context) throws IOException {
         repository.untag(project(projectId), query.docIds, query.tagsAsArray((User)context.currentUser()));
         indexer.untag(project(projectId), query.docIds, query.tagsAsArray((User)context.currentUser()));
         return Payload.ok();
     }
 
-    /**
-     * preflight request
-     *
-     * @param projectId
-     * @param docId
-     * @return 200 PUT
-     */
+    @Operation(description = "preflight request for document untagging",
+            parameters = {
+                    @Parameter(name = "project", description = "the project id", in = ParameterIn.PATH),
+                    @Parameter(name = "docId", description = "document id", in = ParameterIn.PATH)
+            }
+    )
+    @ApiResponse(responseCode = "200", description = "returns PUT")
     @Options("/:project/documents/untag/:docId")
     public Payload untagDocument(final String projectId, final String docId) {return ok().withAllowMethods("OPTIONS", "PUT");}
 
-    /**
-     * Untag one document
-     *
-     * @param projectId
-     * @param docId
-     * @param routing
-     * @param tags
-     * @return 201 if untagged else 200
-     *
-     * $(curl -i -XPUT -H "Content-Type: application/json" localhost:8080/api/apigen-datashare/documents/untag/bd2ef02d39043cc5cd8c5050e81f6e73c608cafde339c9b7ed68b2919482e8dc7da92e33aea9cafec2419c97375f684f -d '["tag1"]')
-     */
+    @Operation(description = "document untagging request",
+            parameters = {
+                    @Parameter(name = "project", description = "the project id", in = ParameterIn.PATH),
+                    @Parameter(name = "docId", description = "document id", in = ParameterIn.PATH),
+                    @Parameter(name = "routing", description = "document routing if not a root document", in = ParameterIn.QUERY)
+            },
+            requestBody = @RequestBody(content = @Content(mediaType = "application/json", schema = @Schema(implementation = List.class)))
+    )
+    @ApiResponse(responseCode = "200", description = "if tag was not in database")
+    @ApiResponse(responseCode = "201", description = "if tag was deleted")
```

Comment:
```suggestion
    @ApiResponse(responseCode = "201", description = "Document had tags; Document tag was deleted")
```
I think 201 is misleading because it's about [resource creation ](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201)